components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -639,4 +659,42 @@ private String getServiceProviderName(String clientId, String tenantDomain) {
         }
         return spName;
     }
+
+    private void handleMissingSessionStateGracefully(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+        String redirectURL = OIDCSessionManagementUtil.getOIDCLogoutURL();
+        String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
+        String postLogoutRedirectUri = request.getParameter(OIDCSessionConstants.OIDC_POST_LOGOUT_REDIRECT_URI_PARAM);
+        if (StringUtils.isEmpty(idTokenHint) || StringUtils.isEmpty(postLogoutRedirectUri)) {
+            response.sendRedirect(getRedirectURL(redirectURL, request));
+            return;
+        }
+        if (!validateIdToken(idTokenHint)) {
+            String msg = "ID token signature validation failed.";
+            log.error(msg);
+            redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
+            response.sendRedirect(getRedirectURL(redirectURL, request));
+            return;
+        }
+        try {
+            String clientId = extractClientFromIdToken(idTokenHint);
+            String callbackUrl = new OAuthAppDAO().getAppInformation(clientId).getCallbackUrl();
+            if (validatePostLogoutUri(postLogoutRedirectUri, callbackUrl)) {
+                redirectURL = postLogoutRedirectUri;
+            } else {
+                redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED,
+                        "Post logout URI does not match with registered callback URI.");
+            }
+        } catch (ParseException e) {
+            String msg = "Error occurred while getting application information.";
+            log.error(msg, e);
+            redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            String msg = "Error occurred while getting application information. Client id not found.";
```

Comment:
Add context information to these logs, to identify which request/client caused the issue

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1130